### PR TITLE
Revert "mount: check right before invoking /bin/umount if it makes sense"

### LINKED ISSUE #28410: Shutdown hangs unmounting /var/lib/nfs/rpc_pipefs
### systemd version the issue has been seen with

254

### Used distribution

_No response_

### Linux kernel version used

6.3.12-200.fc38.x86_64

### CPU architectures issue was seen on

x86_64

### Component

systemd

### Expected behaviour you didn't see

Clean shutdown.

### Unexpected behaviour you saw

Sometimes (but not always) hangs attempting to unmount /var/lib/nfs/rpc_pipefs. Timeout is infinite.

### Steps to reproduce the problem

Presumably only happens on a system using NFS. This system has NFS root and /var is also NFS mounted. Simply run `shutdown -h now` 

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### PR DESCRIPTION
**Pull request by yuwata**
This reverts commit 1483892a421ca34bc841a8e8b1f385744c0407ed.

As the commit says, it does not solve the race. Moreover, it introduces an regression #28410.

Also, checking by `path_is_mount_point()` may trigger automount. From statx(2),

> AT_NO_AUTOMOUNT
>     Don't automount the terminal ("basename") component of pathname if it is a directory that is an automount point.

Similar statements can be found in fstatat(2), which is used in the fallback call for statx() in glibc, and name_to_handle_at(2), which is used as the fallback when statx() failed. So, `path_is_mount_point()` may _do_ trigger automount for parent paths. That should be avoided especially on shutdown.

The original issue #25527 that is 'fixed' by the commit is not serious, and should be fixed by making umount command handle path gracefully: https://github.com/util-linux/util-linux/issues/2132

Fixes #28410.

### CODE REVIEW

**Comment by yuwata** [comment]
The first commit should be backported, if accepted.

File: src/core/mount.c [hunk 1]
```diff
@@ -2069,8 +2069,7 @@ static int mount_process_proc_self_mountinfo(Manager *m) {
 
                 if (!mount_is_mounted(mount)) {
 
-                        /* A mount point is not around right now. It
-                         * might be gone, or might never have
+                        /* A mount point is not around right now. It might be gone, or might never have
```
Comment (keszybz) [inline src/core/mount.c]: I'd prefer to just fold this parts that immediately revert parts of the revert commit into the revert commit, to reduce the churn.

Comment (yuwata) [reply src/core/mount.c]: OK. I will squash them.

**Comment by yuwata** [comment]
> I think we can 'fix' the original issue in other way, but let's do that later in other PR. For stable-release, let's revert the offending commit at first.

Note, I am working on this to mitigate the original issue.

File: src/core/mount.c [hunk 2]
```diff
@@ -195,6 +195,13 @@ static int mount_arm_timer(Mount *m, usec_t usec) {
 
         assert(m);
 
+        if (usec == USEC_INFINITY) {
+                if (m->timer_event_source)
+                        return sd_event_source_set_enabled(m->timer_event_source, SD_EVENT_OFF);
+
+                return 0;
```
Comment (keszybz) [inline src/core/mount.c]: One nitpick: `sd_event_source_set_enabled()` is a noop if `m == SD_EVENT_OFF && !s`. So please collapse this to:
```C
if (usec == USEC_INFINITY)
   return sd_event_source_set_enabled(m->timer_event_source, SD_EVENT_OFF);
```

**Review by keszybz — APPROVED** [review]
Since this is a revert for an patch that was supposed to fix the an issue, but doesn't actually fix it, but causes a regression instead, we should lean towards merging, even if the whole story is not clear yet. If instead this was a new feature, we would have be more cautious.

File: src/core/mount.c (hunk 2 above)
Comment (keszybz) [reply src/core/mount.c]: I force-pushed with this change. I hope that's OK.

**Review by bluca — APPROVED** [review]

File: src/core/mount.c (hunk 2 above)
Comment (yuwata) [reply src/core/mount.c]: Thanks.

**Comment by keszybz** [comment]
I still think we should revert something that is broken, if we don't immediately have a fix available. Reverting the revert is a trivial, so if the issue is easy to fix, then the revert does not make that harder. But we'll have a clearer fix if it's all done as part of one pull request.

**Pull request merged**